Add keyword marker token filter docs #8065 (#8134)
opensearch-project/documentation-website: _analyzers/token-filters/keyword-marker.md

---
layout: default
title: Keyword marker
parent: Token filters
nav_order: 200
---

# Keyword marker token filter

The `keyword_marker` token filter is used to prevent certain tokens from being altered by stemmers or other filters. The `keyword_marker` token filter does this by marking the specified tokens as `keywords`, which prevents any stemming or other processing. This ensures that specific words remain in their original form. 

## Parameters

The `keyword_marker` token filter can be configured with the following parameters.

Parameter | Required/Optional | Data type | Description
:--- | :--- | :--- | :--- 
`ignore_case` | Optional | Boolean | Whether to ignore the letter case when matching keywords. Default is `false`.

> `keywords` | Required if `keywords_path` or `keywords_pattern` is not set | List of strings | List of tokens to mark as keywords. 

Reviewer: ```suggestion
`keywords` | Required if either `keywords_path` or `keywords_pattern` is not set | List of strings | The list of tokens to mark as keywords. 
```

> `keywords_path` | Required if `keywords` or `keywords_pattern` is not set | String | Path (relative to the `config` directory or absolute) to the list of keywords.

Reviewer: ```suggestion
`keywords_path` | Required if either `keywords` or `keywords_pattern` is not set | String | The path (relative to the `config` directory or absolute) to the list of keywords.
```

> `keywords_pattern` | Required if `keywords` or `keywords_path` is not set | String | [Regular expression](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html) used for matching tokens to be marked as keywords.

Reviewer: ```suggestion
`keywords_pattern` | Required if either `keywords` or `keywords_path` is not set | String | A [regular expression](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html) used for matching tokens to be marked as keywords.
```

 

## Example

> The following example request creates a new index named `my_index` and configures an analyzer with `keyword_marker` filter. The filter marks the word `example` as a keyword:

Reviewer: ```suggestion
The following example request creates a new index named `my_index` and configures an analyzer with a `keyword_marker` filter. The filter marks the word `example` as a keyword:
```

```json
PUT /my_index
{
  "settings": {
    "analysis": {
      "analyzer": {
        "custom_analyzer": {
          "type": "custom",
          "tokenizer": "standard",
          "filter": ["lowercase", "keyword_marker_filter", "stemmer"]
        }
      },
      "filter": {
        "keyword_marker_filter": {
          "type": "keyword_marker",
          "keywords": ["example"]
        }
      }
    }
  }
}
```
{% include copy-curl.html %}

## Generated tokens

Use the following request to examine the tokens generated using the analyzer:

```json
GET /my_index/_analyze
{
  "analyzer": "custom_analyzer",
  "text": "Favorite example"
}
```
{% include copy-curl.html %}

> The response contains the generated tokens. Note that the word `favorite` was stemmed but the word `example` was not stemmed because it was marked as a keyword:

Reviewer: ```suggestion
The response contains the generated tokens. Note that while the word `favorite` was stemmed, the word `example` was not stemmed because it was marked as a keyword:
```

---
layout: default
title: Keyword marker
parent: Token filters
nav_order: 200
---

# Keyword marker token filter

> The `keyword_marker` token filter in OpenSearch is used to prevent certain tokens from being altered by stemmers or other filters. This ensures that specific words remain in their original form. `keyword_marker` token filter does this by marking tokens as `keywords` which prevents any stemming or other processing.

Reviewer 2: ```suggestion
The `keyword_marker` token filter is used to prevent certain tokens from being altered by stemmers or other filters. The `keyword_marker` token filter does this by marking the specified tokens as `keywords`, which prevents any stemming or other processing. This ensures that specific words remain in their original form. 
```

## Parameters

> The `keyword_marker` token filter in OpenSearch can be configured with the following parameters.

Reviewer 2: ```suggestion
The `keyword_marker` token filter can be configured with the following parameters.
```

Parameter | Required/Optional | Data type | Description
:--- | :--- | :--- | :--- 

> `ignore_case` | Optional | Boolean | Ignore the letter case when matching keywords. Default is `false.`

Reviewer 2: ```suggestion
`ignore_case` | Optional | Boolean | Whether to ignore the letter case when matching keywords. Default is `false`.
```

> `keywords` | Required if `keywords_path` or `keywords_pattern` is not set | List of strings | List of strings used to match tokens.

Reviewer 2: ```suggestion
`keywords` | Required if `keywords_path` or `keywords_pattern` is not set | List of strings | List of tokens to mark as keywords. 
```

> `keywords_path` | Required if `keywords` or `keywords_pattern` is not set | String | Path (relative to `config` directory or absolute) to list of strings.

Reviewer 2: ```suggestion
`keywords_path` | Required if `keywords` or `keywords_pattern` is not set | String | Path (relative to the `config` directory or absolute) to the list of keywords.
```

> `keywords_pattern` | Required if `keywords` or `keywords_path` is not set | String | [Regular expression](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html) used for matching tokens.

Reviewer 2: ```suggestion
`keywords_pattern` | Required if `keywords` or `keywords_path` is not set | String | [Regular expression](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html) used for matching tokens to be marked as keywords.
```

 

## Example

> The following example request creates a new index named `my_index` and configures an analyzer with `keyword_marker` filter:

Reviewer 2: Not sure if OpenSearch is the best word for showcasing the filter. I would shorten the sentence to a couple of words which are normally stemmed. Then show examples with and without the filter so users can see that the filter prevents stemming.
